### ISSUE TITLE
Feature Request: DNSCrypt support

### ISSUE THREAD

**Issue by muety** [prometheus/blackbox_exporter#696]
I don't know how easy this can be accomplished. But after having support for DNS-over-TLS (#643) probes, it might be convenient to also be able to probe [DNSCrypt](https://dnscrypt.info/protocol) endpoints.

**Comment by brian-brazil** [comment]
I've not heard of this, and it looks like it was never submitted to the IETF which is a bit odd. Does the DNS library we use support this?

**Comment by muety** [comment]
I'm not an expert myself, but there are [discussion](https://www.reddit.com/r/privacy/comments/89pr15/dnsoverhttps_vs_dns_overtls_vs_dnscrypt/dwsosjr/?utm_source=share&utm_medium=web2x) about DNSCrypt vs. DoT. Also, there's the widely used [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy). However, as you already pointed out, DNSCrypt has not made it to an official standard, yet.

> Does the DNS library we use support this?

There was short [discussion](https://github.com/miekg/dns/issues/297) about it, but apparently nobody took the initiative to implement it, yet. I assume that's a dealbreaker for blackbox exporter to support it?

**Comment by brian-brazil** [comment]
> I assume that's a dealbreaker for blackbox exporter to support it?

Yes, what is essentially application-specific code doesn't belong in this exporter. It's intended to cover very basic standard stuff.

**Comment by electron0zero** [comment]
We have DNS over TLS support, added in https://github.com/prometheus/blackbox_exporter/pull/643.

I am going to close this because DNSCrypt is not an official standard, and less adopted then DoT or DoH.
